k9mail/src/main/java/com/fsck/k9/mailstore/LockableDatabase.java

```diff
@@ -380,6 +372,7 @@ private void openOrCreateDataspace() throws UnavailableStorageException {
             } catch (SQLiteException e) {
                 // try to gracefully handle DB corruption - see issue 2537
                 Log.w(K9.LOG_TAG, "Unable to open DB " + databaseFile + " - removing file and retrying", e);
+                // noinspection ResultOfMethodCallIgnored, not much we can do if this fails
```

Comment:
We could throw an exception instead of letting `doOpenOrCreateDb()` fail with a possibly unhelpful error message.